### DISCUSSION TITLE
ESP32-S3 Bluetooth

### DISCUSSION THREAD

**Discussion by josef2600** [pschatzmann/arduino-audio-tools#1697] [Q&A] [answered]
i know it doesn't support A2DP API or anything related to using Bluetooth audio now, and that it is not supporting aptX at all.
but, i am thinking about making a great audio Bluetooth with support of network. so far, the ESP32-S3 has the best performance. 
let me make it short! since i didn't get anything from hardware in datasheet, i wanted to ask you this. does Bluetooth audio has a dedicated hardware in ESP32 that is not in ESP32-S3? or it is just from the programming point of view?
thank you.

**Comment by pschatzmann** [comment, marked as answer]
Just read the technical sheet: you need to look for support of Bluetooth classic. 
So the S3 is missing the relevant hardware support!

You find plenty of other examples in the [examples-communications folder](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication)

**Reply by josef2600** [reply]
thanks. i know, you have said it doesn't support it a lot! i was just wondering. now i know it doesn't have the hardware needed for it.

**Reply by pschatzmann** [reply]
Quite some time ago, I was reading that Espressif was planning to provide audio support for BLE.
I haven't heard anything about this since then...

**Reply by malaki86** [reply]
> Quite some time ago, I was reading that Espressif was planning to provide audio support for BLE. I haven't heard anything about this since then...

The new ESP32-H4 is supposed to have Low-energy audio

**Comment by josef2600** [comment]
another question, some say SBC and SBC-XQ have the latency over 500ms, and i see it too, but some say it can be as low as 67ms. do they mean with low quality? i am a bit confused.

**Reply by pschatzmann** [reply]
The latency is quite bad: this means that it takes quite some time after you sent it until it is played back on the audio sink.
This is bad e.g. if you want to implement some kind of instrument.
The quality is pretty good however

**Reply by josef2600** [reply]
i guess my only option is to buy their specific chips. it is a problem. i sow the delay for the first time when i was using it to watch a movie. i was confused what is wrong! personally i spent a lot on an ear pod and it was shit! so i decided to make a perfect one! i have been developing a class A amp for a headset. i just love weird things to do!
i have seen few chips, but don't know whats what. but i am searching for a decoder chip without micro, since i want esp32 to do many things too. 
thank you.
